### ISSUE TITLE
Screen reader: pronounce note stem directions and note group boundaries

### ISSUE THREAD

**Issue by VladislavBuyalski** [musescore/MuseScore#23941]
### Your idea

1. We would be glad to hear the directions of the note stem so that if an incorrect direction occurs, we can correct it.
2. Also, we would be glad to hear the boundaries of the note group in order to correct them ourselves or create our own note groups in complex time signatures. For example: 5/4 or 7/8 and so on.

### Problem to be solved

These ideas will help blind musicians to control the directions of note stem and note groups independently. And also save time for sighted musicians who help blind musicians to design notes.

### Checklist

- [X] This request follows the [guidelines](https://github.com/musescore/MuseScore/wiki/Reporting-bugs-and-issues) for reporting issues
- [X] I have verified that this feature request has not been logged before, by searching the [issue tracker](https://github.com/musescore/MuseScore/issues) for similar requests

**Comment by shoogle** [comment]
Good suggestions!

1. I think it would be sensible to say the stem direction only when it changes, so if there are several "stem up" notes in a row, you'd only hear it on the first one.

    If there are overlapping voices, we could say when the stem direction changes compared to a note that was previously selected in the _same voice_, or in _any voice_. I think the latter is less confusing, but maybe this should configurable in Preferences.

2. As you say, we only need to hear the beaming on group boundaries, so "start beam" and "end beam". Possibly just "start beam" is enough, since we know it will end at the next "start beam", quarter note, half note, whole note, barline, or rest. If the beam unexpectedly continues (e.g. over a rest) we can say "continue beam".

    It might also be worth announcing subgroupings, such as where 2 or more beams temporarily become 1 beam to indicate (sub-)beat boundaries.

![image](https://github.com/user-attachments/assets/12b1eea4-6fef-4af3-9781-676adfed2b84)

**Comment by VladislavBuyalski** [comment]
1. I agree that the stem direction should be pronounced only when it changes. For example: if several notes have the same direction, then we hear the direction of only the first note in this group. If in the next group the direction changes, we also hear only the direction of the first note.
If we are talking about a chord, we also hear the stem direction on any of these notes in the chord, because the chord has one stem in one direction. When we move between the notes of the chord, we do not need to repeat the stem direction.
If we have 2, 3 or 4 voices, as in polyphonic music, I think that when we move between voices we should hear both the voice number that we already have and the stem direction for that voice.
2. I completely agree with your comment.
For example: If we have a complex time signature of 7/8, we create a group of 4 eighth notes + 3 eighth notes, we should hear only the beginning of the beam for each of these groups.
Example 2: If in the first group of notes the first two eighth notes are divided into sixteenth notes, we should hear the beginning of the common beam and the beginning of the beam for the group of sixteenth notes, after the end of the subgroup, we should hear the continuation of the common beam for this group of notes. Also for thirty-second notes and various multiples (triplet, quintuplet, etc.). This way we will be able to understand the boundaries of additional subgroups within one group of notes.
Example 3: If in the first group the third note looks like a rest, then at this rest we should hear the continuation of the common beam for this group of notes. This way we will be able to understand that the group of notes has not yet ended.
